.github/workflows/e2e.yaml

```diff
@@ -0,0 +1,110 @@
+---
+name: E2E Tests
+
+on:
+  pull_request:
+    types:
+      - opened
+      - reopened
+      - synchronize
+    paths-ignore:
+      - .circlecli/**
+      - meta/**
+      - .gitignore
+      - .markdownlint.yml
+      - .yamllint
+      - ansible.cfg.galaxy
+      - CHANGELOG.md
+      - LICENSE
+      - poetry.lock
+      - pyproject.toml
+      - README.md
+      - Vagrantfile
+  push:
+    branches:
+      - master
+      - ansible-molecule-4
+  schedule:
+    - cron: "0 1 * * *"
+  workflow_dispatch:
+
+env:
+  PY_COLORS: true
+  ANSIBLE_FORCE_COLOR: true
+  MOLECULE_DEBUG: false
+  MOLECULE_VERBOSITY: 0
+  MOLECULE_NO_LOG: true
+
+concurrency:
+  group: e2e
+  cancel-in-progress: false
+
+jobs:
+  molecule:
+    runs-on: ubuntu-20.04
+    strategy:
+      max-parallel: 2
+      fail-fast: false
+      matrix:
+        st2_repo:
+          - stable
+          - unstable
+        os:
+          - centos7
+          - ubuntu18
+          - rockylinux8
+          - ubuntu20
+        python:
+          # - version: 3.6
+          #   ansible-core: 2.11.12
+          #   molecule: 3.6.1
+          #   molecule-docker: 1.1.0
+          #   molecule-scenario: python36
+          - version: 3.8
+            ansible-core: 2.11.12
+            molecule: 3.6.1
+            molecule-docker: 1.1.0
+            molecule-scenario: python38
+
+    steps:
+      - name: Checkout the repository
+        uses: actions/checkout@v3
+
+      - name: Set up Python ${{ matrix.python.version }}
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python.version }}
+
+      - name: Install Python dependencies
+        run: |
+          python3 -m pip install --upgrade pip
+          python3 -m pip install --upgrade ansible-core==${{ matrix.python.ansible-core }}
+          python3 -m pip install --upgrade molecule==${{ matrix.python.molecule }}
+          python3 -m pip install --upgrade molecule-docker==${{ matrix.python.molecule-docker }}
```

Comment:
Is there a docker image with ansible (and maybe molecule?) already pre-installed?
as said, we can use any version of molecule, only the ansible core version is important in these tests.

Reply:
I'm not aware of one, I'll look around a bit.

Reply:
Haven't had much luck finding one, in any event, setting up Python and the requirements takes about a minute.